### PR TITLE
made 4 different components for the editprofile section

### PR DESCRIPTION
**Pull request by antonio-b**
@mikhael28 
exclude the projects.js file, that was sent in error. I separated the components. I wanted to get your approval on this before i moved on toward fixing the bullet points. thanks

### CODE REVIEW

**Comment by mikhael28** [comment]
Hi @antonio-b this is great - would you please reduce the state objects in each of the components, to only contain the essential form/loading state required for that component to work? Right now, you are duplicating a lot of state. Also, make sure to update the 'toast' notifications in the try catch block to have context sensitive messages.

**Comment by mikhael28** [comment]
What do you mean by 'fixing bullet points'?

**Comment by antonio-b** [comment]
Yes, I'm seeing the state objects issue now because of the merge conflict. I thought I reduced it but not enough.

**Comment by mikhael28** [comment]
Make sure to pull the latest code from the upstream repository - you may have to manually respond to merge conflicts. These are huge pains and headaches to deal with at first, but they will become common and simple soon enough in your career :)

File: src/profile/EditProfile.js [hunk 1]
```diff
@@ -215,14 +324,178 @@ export default class EditProfile extends Component {
...
         </div>
        <Button onClick={() => this.props.history.push("/settings/password")}>
+
+
+        <br />
+
+        {/* Here we are updating our default language */}
+
+        <FormGroup controlId="defaultLanguage" bsSize="large">
+          <ControlLabel>Default Language</ControlLabel>
```
Comment (mikhael28) [inline src/profile/EditProfile.js]: It looks like you need to add back in the 4 form components here!

**Review by mikhael28 — COMMENTED** [review]
I would mark this as a 'draft pull-request', until you are ready for a potentially final review of the code, pre-merge.

**Comment by antonio-b** [comment]
1.saving the default language when the user makes a selection and showing a loading indicator - instead of relying on an ugly, mis-styled button.
2.Ability to edit/update GitHub username
3.Loading indicator when creating Projects
4.Editing your projects (and viewing them somewhere, in another component)

These are the bullet points i was referring to.

**Comment by antonio-b** [comment]
I just read the comment about making it a draft p.r after i submitted my reworked components.

**Pull request closed without merging**